### PR TITLE
ci-operator: add optional, best-effort reporting of result

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @ingvagabund @alvaroaleman

### CODE REVIEW

File: pkg/results/report.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+type Options struct {
+	address  string
+	certFile string
+	keyFile  string
+	caFile   string
+}
+
+// Bind adds flags for the options
+func (o *Options) Bind(flag *flag.FlagSet) {
+	flag.StringVar(&o.address, "report-address", "", "Address of the aggregate reporting server.")
```
Comment (alvaroaleman) [inline pkg/results/report.go:29]: Why? We can just use a specific log line or even junit and find it via ci-search?

Comment (alvaroaleman) [reply pkg/results/report.go:29]: If we are building an aggregation service I'd like to see some design first

Comment (stevekuznetsov) [reply pkg/results/report.go:29]: Design has been hashed out pretty well at this stage.

Comment (petr-muller) [reply pkg/results/report.go:29]: Where?

File: pkg/results/report.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+type Request struct {
+	// JobName is the name of the job for which a result is being reported
+	JobName string `json:"job_name"`
+	// Type is the type of job ("presubmit", "postsubmit", "periodic" or "batch")
+	Type string `json:"type"`
+	// State is "succeeded" or "failed"
+	State string `json:"state"`
+	// Reason is a colon-delimited list of reasons for failure
+	Reason string `json:"reason"`
+}
```
Comment (ingvagabund) [inline pkg/results/report.go:93]: Cluster field is missing

Comment (stevekuznetsov) [reply pkg/results/report.go:93]: Yeah, I thought about that one -- harder to do. Will attempt to do better now.

**Pull request merged**